_posts/2020-05-15-model-inversion-attacks/model_inversion_differential_privacy.Rmd

```diff
@@ -0,0 +1,740 @@
+---
+title: "Hacking deep learning: model inversion attack by example"
+description: > 
+  Compared to other applications, deep learning models might not seem too likely as victims of privacy attacks. However, methods exist to determine whether an entity was used in the training set (an adversarial attack called member inference), and techniques subsumed under "model inversion" allow to spy on raw data input given just model output (and sometimes, context information). This post shows an end-to-end example of model inversion, and explores mitigation strategies using TensorFlow Privacy. 
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanamodelinversion
+bibliography: bibliography.bib
+date: 05-15-2020
+categories:
+  - R
+  - Privacy & Security
+  - TensorFlow/Keras
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/results.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+
+How private are individual data in the context of machine learning models? The data used to train the model, say. There are
+types of models where the answer is simple. Take k-nearest-neighbors, for example. There *is* not even a model without the
+complete dataset. Or support vector machines. There is no model without the support vectors. But neural networks? They're just
+some composition of functions, -- no data included.
+
+Same with data fed to a deployed deep learning model. It's pretty unlikely one could invert the final softmax output from a
+big ResNet and get back the raw input data.
+
+In theory, then, "hacking" a standard neural net to spy on input data sounds illusory. In practice, however, there is always
+some real-world *context*. The context may be other datasets, publicly available, that can be linked to the "private" data in
+question. This is a popular showcase used in advocating for differential privacy[@Dwork2006]: Take an "anonymized" dataset,
+dig up complementary information from public sources, and de-anonymize records ad libitum. Some context in that sense will
+often be used in "black-box" attacks, ones that presuppose no insider information about the model to be hacked.
+
+But context can also be structural, such as in the scenario demonstrated in this post. For example, assume a distributed
+model, where sets lof layers run on different devices (embedded devices or mobile phones, for example). (A scenario like that
```

Comment:
```suggestion
model, where sets of layers run on different devices (embedded devices or mobile phones, for example). (A scenario like that
```

Reply:
Having a parenthetical right after a sentence that ends with a parenthetical just feels off to me.